modules/local/sratools/fasterqdump/main.nf

```diff
@@ -22,35 +22,32 @@ process SRATOOLS_FASTERQDUMP {
     script:
     def args = task.ext.args ?: ''
     def args2 = task.ext.args2 ?: ''
-    def args3 = task.ext.args3 ?: ''
-    def prefix = task.ext.prefix ?: "${meta.id}"
+    def prefix = (meta.id && task.ext.add_prefix) ? "${meta.id}_" : ""
     def key_file = ''
-
-    if (certificate.toString().endsWith('.jwt')){
+    if (certificate.toString().endsWith('.jwt')) {
         key_file += " --perm ${certificate}"
-        }
-    else if (certificate.toString().endsWith('.ngc')){
+    } else if (certificate.toString().endsWith('.ngc')) {
         key_file += " --ngc ${certificate}"
     }
-
     """
     export NCBI_SETTINGS="\$PWD/${ncbi_settings}"
 
     # Make directory ahead of time since otherwise
     # fasterq-dump does not set correct permissions/owner
-    mkdir -p reads
+    mkdir -p reads_noprefix reads
 
     fasterq-dump \\
         $args \\
         --threads $task.cpus \\
-        --outdir reads \\
+        --outdir reads_noprefix \\
         ${key_file} \\
         ${sra}
 
-    find reads/ -type f   -name "$args2" -exec mv {} {}.fastq \\;
+    # Adds {prefix} to the output fastq file names
+    for i in reads_noprefix/*.fastq; do b=`basename "\${i}"`; mv "reads_noprefix/\${b}" "reads/${prefix}\${b}"; done
```

Comment:
Just curious why `mv "reads_noprefix/\${b}"` instead of just `mv \$i`. 

Not that it really makes a difference in execution

Reply:
Both would work. Just a preference thing. That is, I like seeing written out which is the source and destination directory in the `mv` command.